### PR TITLE
Feature #428/cache from url

### LINKED ISSUE #428: API to instantiate a cache from a resource url
From https://github.com/encode/starlette/issues/252.

Let's add an API to create caches from resource urls. First thing that comes to my mind is add support in `caches.create` call to do that.

### PR DESCRIPTION
**Pull request by argaen**
This PR adds the `Cache.from_url` method to instantiate caches from a specific resource url.

Couple of things:
- password is passed through queryparams. According to IANA proposal password is passed as the user info param. However user doesn't make sense because Redis authentication only goes with password.
- maybe it would make sense for redis to have the `db` as a path param instead of having it as a query param. However its the only one so I think its better to have it as queryparam like all the other things you can configure that then get propagated to the underlying client

Docs:
screenshot from 2019-01-05 21-53-47

Closes #428

### CODE REVIEW

File: aiocache/backends/memcached.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def __init__(self, endpoint="127.0.0.1", port=11211, pool_size=2, loop=None, **k
         super().__init__(**kwargs)
         self.endpoint = endpoint
         self.port = port
-        self.pool_size = pool_size
+        self.pool_size = int(pool_size)
```
Comment (argaen) [inline aiocache/backends/memcached.py]: unrelated with PR but querystring from `from_url` method always parses args as strings. This is helpful anyway as a standalone feature

**Comment by tomchristie** [comment]
> maybe it would make sense for redis to have the db as a path param instead of having it as a query param. However its the only one so I think its better to have it as queryparam like all the other things you can configure that then get propagated to the underlying client

Indeed. Other versions I've seen of redis URLs treat the path as the database component, which is consistent with eg. how database URLs are used. 

So: `redis://[:password@]host[:port][/db-number][?option=value]`

Eg: see https://www.npmjs.com/package/redis-url

**Comment by tomchristie** [comment]
Also *provisionally* registered with IANA: https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml

**Comment by argaen** [comment]
> Indeed. Other versions I've seen of redis URLs treat the path as the database component, which is consistent with eg. how database URLs are used.

Yeah and aioredis also accepts that. I was just being lazy because this means implementation of this feature will be backend specific then :(. Will update 😄

**Comment by argaen** [comment]
Updated, @tomchristie let me know what you think :)

**Comment by tomchristie** [comment]
Looks great!

**Pull request merged**